File: Q3.cpp
#include<iostream>
#include<fstream>
#include<string>
#include<vector>
using namespace std;
vector<vector<float>> store(string st){
    vector<vector<float>> X;
    vector<float> U;
    float T[3];
    ifstream file(st);
    if (file.is_open()){
       for(int i=0;i<3;i++){ 
            for(int i=0;i<3;i++){
                file>>T[i];
            }
            for(int i=0;i<3;i++){
                U.push_back(T[i]);
            }
            X.push_back(U);
            U={};
       }
    }
    return X;

}
int main(){
    vector<vector<float>> M=store("testA.txt");//This is where you enter the name of the text file
    vector<vector<float>> N=store("testB.txt");
    cout<<"The first Matrix is M: ";
    for(int i=0;i<3;i++){
        if(i>0) {
            cout<<endl;
            cout<<"                       ";
        }
        for(int j=0; j<3;j++){
            cout<<M[i][j]<<"   ";
        }
    }
    cout<<endl;
    cout<<endl;
    cout<<"The second Matrix is N: ";
    for(int i=0; i<3;i++){
        if(i>0) {
            cout<<endl;
            cout<<"                        ";
        }
        for(int j=0; j<3;j++){
            cout<<N[i][j]<<"   ";
        }
    }
    vector<vector<float>> prod={{},{},{}};
    float t=0;
    for (int i=0;i<3;i++){
        for (int j=0;j<3;j++){
            for (int k=0;k<3;k++){
                t+=M[i][k]*N[k][j];
            }
            prod[i].push_back(t);
            t=0;
        }  
    }
    cout<<endl;
    cout<<endl;
    cout<<"The product of the two Matrices is MxN: ";
    for(int i=0; i<3;i++){
        if(i>0) {
            cout<<endl;
            cout<<"                                        ";
        }
        for(int j=0; j<3;j++){
            cout<<prod[i][j]<<"   ";
        }
    }
    vector<float> A={3,4,5};
    vector<float> prod2;
    float s=0;
    for (int l=0;l<3;l++){
        for (int k=0;k<3;k++){
            s+= M[l][k]*A[k];
        }
        prod2.push_back(s);
        s=0;
    }
    cout<<"\nThe product of M and vector ";
    for (int i=0;i<4;i++){
        if (i==0) cout<<"(";
        else {
            if (i!=3) cout<<A[i-1]<<",";
            else cout<<A[i-1]<<")";
        }
    } 
    cout<<" is: \n";
    for (int i=0;i<4;i++){
        if (i==0) cout<<"(";
        else {
            if (i!=3) cout<<prod2[i-1]<<",";
            else cout<<prod2[i-1]<<")";
        }
    }
}
/* M=1 0 0      N=0 1 0
     0 0 1        0 0 1
     0 1 0        1 0 0

prod(MxN)=0 1 0 
     1 0 0
     0 0 1

A=(3,4,5) => MxA= (3,5,4)*/

File: Q3.py
def store(st):
    X=[]
    T=[]
    with open(st,'r+') as file:
        i=0
        while i<3:
            s=file.readline()
            p=s.split()
            j=0
            while j<3:
                T.append(float(p[j]))
                j=j+1
            X.append(T)
            T=[]
            i=i+1 
        return X 
#This is where you enter the name of the text file
'''txt_M=input(str("Enter the name of the file containing the first matrix "))+str(".txt")#for input textfile name
txt_N=input(str("Enter the name of the file containing the second matrix "))+str(".txt")
M=store(txt_M)
N=store(txt_N)'''
M=store("testA.txt")
N=store("testB.txt")
prod=[[],[],[]]
t=0
for i in range(3):
    for j in range(3):
        for k in range(3):
            t+=M[i][k]*N[k][j]
        prod[i].append(t)
        t=0
    j=0
print("The first Matrix is M: ")
for i in range(3):
    if(i>0):
        print('\n')
    for j in range(3):
        print(M[i][j],end="   ")
print('\n')
print("The second Matrix is N: ")
for i in range(3):
    if(i>0):
        print('\n')
    for j in range(3):
        print(N[i][j],end="   ")
print('\n')
print("The product of the two Matrices is MxN: ")
for i in range(3):
    if(i>0):
        print('\n')
    for j in range(3):
        print(prod[i][j],end="   ")
print('\n')
A=[3,4,5]
prod2=[]
s=0
for l in range(3):
    for k in range(3):
        s+= M[l][k]*A[k]
    prod2.append(s)
    s=0
print("The product of M and vector {0} is :\n{1}".format(A,prod2))
''' M=1 0 0      N=0 1 0
     0 0 1        0 0 1
     0 1 0        1 0 0

prod(MxN)=0 1 0 
     1 0 0
     0 0 1

A=(3,4,5) => MXA= (3,5,4)'''